compiler/rustc_mir/src/borrow_check/diagnostics/mutability_errors.rs

```diff
@@ -424,15 +424,108 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
 
                         match label {
                             Some((true, err_help_span, suggested_code)) => {
-                                err.span_suggestion(
-                                    err_help_span,
-                                    &format!(
-                                        "consider changing this to be a mutable {}",
-                                        pointer_desc
-                                    ),
-                                    suggested_code,
-                                    Applicability::MachineApplicable,
-                                );
+                                /// User cannot make signature of a trait mutable
+                                /// without changing the trait. So we find if this
+                                /// error belongs to a trait and if so we move
+                                /// suggestion to the trait or disable it if it is
+                                /// out of scope of this crate
+                                let (is_trait_sig, local_trait) = {
+                                    if self.body.local_kind(local) != LocalKind::Arg {
+                                        (false, None)
+                                    } else {
+                                        let hir_map = self.infcx.tcx.hir();
+                                        let my_hir = hir_map.local_def_id_to_hir_id(
+                                            self.body.source.def_id().as_local().unwrap(),
+                                        );
+                                        match hir_map.find(hir_map.get_parent_node(my_hir)) {
```

Comment:
You can use `tcx.opt_associated_item` to determine if the method comes from a trait impl.

Reply:
I also used `my_hir` in `hir_map.name(hi) != hir_map.name(my_hir)` as well. If problem is count of characters, I don't think `tcx.opt_associated_item` help much?